File: 1.Two_Sum/1.Two_Sum.scala
import scala.collection.mutable.HashMap

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var hash = new HashMap[Int, Int]
        for (i <- 0 until nums.size) {
            if (hash.contains(target - nums(i)))
                return Array(hash(target - nums(i)), i)
            else 
                hash += (nums(i)->i)
        }
        return Array();
    }
}

File: 1.Two_Sum/1.Two_Sum.go
func twoSum(nums []int, target int) []int {
    hash :=  make(map[int]int)
    for i := 0; i < len(nums); i++ {
        if _, ok := hash[target - nums[i]]; ok {
            return []int{hash[target - nums[i]], i}
        } else {
            hash[nums[i]] = i
        }
    }
    return []int{}
}
